src/ldp/http/SimpleBodyParser.ts

```diff
@@ -0,0 +1,52 @@
+import { BodyParser } from './BodyParser';
+import { HttpRequest } from '../../server/HttpRequest';
+import { Quad } from 'rdf-js';
+import { QuadRepresentation } from '../representation/QuadRepresentation';
+import { RepresentationMetadata } from '../representation/RepresentationMetadata';
+import { StreamParser } from 'n3';
+import { TypedReadable } from '../../util/TypedReadable';
+import { UnsupportedMediaTypeHttpError } from '../../util/errors/UnsupportedMediaTypeHttpError';
+import 'jest-rdf';
+
+export class SimpleBodyParser extends BodyParser {
+  private static readonly contentTypes = [
+    'application/n-quads',
+    'application/trig',
+    'application/n-triples',
+    'text/turtle',
+    'text/n3',
+  ];
+
+  public async canHandle(input: HttpRequest): Promise<void> {
+    const contentType = input.headers['content-type'];
+
+    if (contentType && !SimpleBodyParser.contentTypes.some((type): boolean => contentType.includes(type))) {
+      throw new UnsupportedMediaTypeHttpError('This parser only supports RDF data.');
+    }
+  }
+
+  public async handle(input: HttpRequest): Promise<QuadRepresentation> {
+    const contentType = input.headers['content-type'];
+
+    if (!contentType) {
+      return undefined;
+    }
+
+    const specificType = contentType.split(';')[0];
```

Comment:
```suggestion
    const mediaType = contentType.split(';')[0];
```

Note that a content type does not always have a `;`.

Reply:
Yea if it has no `;` the result of the split will be the entire string

Reply:
Might also have comma-separated params